sdk/storage/azure-storage-blob/azure/storage/blob/download.py

```diff
@@ -371,6 +398,14 @@ def _initial_request(self):
             else:
                 process_storage_error(error)
 
+        # get page ranges to optimize downloading sparse page blob
+        if response.properties.blob_type == 'PageBlob':
+            try:
+                page_ranges = self.clients.page_blob.get_page_ranges()
+                self.non_empty_ranges = get_page_ranges_result(page_ranges)[0]
+            except HttpResponseError:
```

Comment:
>HttpResponseError [](start = 19, length = 17)

add comment to explain the nuances of a timeout: it means the blob is not sparse enough